util/wait_group_wrapper.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"sync"
+
+	"github.com/pingcap/tidb/util/logutil"
+	"go.uber.org/zap"
+)
+
+// WaitGroupWrapper is a wrapper for sync.WaitGroup
+type WaitGroupWrapper struct {
+	sync.WaitGroup
+}
+
+// Run run a function in a goroutine, add 1 to WaitGroup
+// and call done when function return. Please not use panic
+// in the cb function.
+func (w *WaitGroupWrapper) Run(cb func()) {
+	w.Add(1)
+	go func() {
+		cb()
+		w.Done()
```

Comment:
BTW, I guess kenny means call `w.Done()` by `defer w.Done()`(which would be executed even `cb` panics).

Reply:
OK, I have fixed.